Source/Libraries/openXDA.Nodes/Types/Analysis/AnalysisNode.cs

```diff
@@ -306,19 +312,51 @@ private void Process(MeterDataSet meterDataSet)
                     }
 
                     Log.Debug($"Finished executing data operation '{dataOperationDefinition.UnqualifiedTypeName}'.");
+                    if (index == 0) status = FileGroupProcessingStatus.Success;
+                    else if (status == FileGroupProcessingStatus.Failed) status = FileGroupProcessingStatus.PartialSuccess;
                 }
                 catch (Exception ex)
                 {
                     // Log the error and skip to the next data operation
                     string message = $"An error occurred while executing data operation of type '{dataOperationDefinition.TypeName}' on data from meter '{meterDataSet.Meter.AssetKey}': {ex.Message}";
                     Exception wrapper = new Exception(message, ex);
+                    if (status == FileGroupProcessingStatus.Success) status = FileGroupProcessingStatus.PartialSuccess;
                     Log.Error(wrapper.Message, wrapper);
+                    dataOperationFailures.Add(new DataOperationFailure
+                    {
+                        DataOperationID = dataOperationDefinition.ID,
+                        FileGroupID = meterDataSet.FileGroup.ID,
+                        Log = message,
+                        StackTrace = ex.StackTrace,
+                        TimeOfFailure = DateTime.UtcNow
+                    });
+                    if (IsDisposed)
+                    {
+                        LogDataOperationFailures(dataOperationFailures);
+                        if (index == 0) return FileGroupProcessingStatus.Failed;
+                        return status;
+                    }
                 }
+
+                index++;
             }
 
-            FileGroup fileGroup = meterDataSet.FileGroup;
```

Comment:
removing that seems problematic - though it could be a result of Github not aligning changes well. Is that still in there somewhere

Reply:
It got moved up into line 275